src/main.cpp

```diff
@@ -4179,25 +4298,25 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                     }
                 }
             }
-            else if (inv.IsKnownType())
+            else if (inv.type == MSG_TX || inv.type == MSG_WITNESS_TX)
             {
                 // Send stream from relay memory
                 bool pushed = false;
                 {
-                    LOCK(cs_mapRelay);
-                    map<CInv, CDataStream>::iterator mi = mapRelay.find(inv);
-                    if (mi != mapRelay.end()) {
-                        pfrom->PushMessage(inv.GetCommand(), (*mi).second);
+                    LOCK(cs_mapRelayTx);
+                    map<uint256, CTransaction>::iterator mi = mapRelayTx.find(inv.hash);
+                    if (mi != mapRelayTx.end()) {
+                        pfrom->PushMessageWithFlag(inv.type == MSG_WITNESS_TX ? SERIALIZE_TRANSACTION_WITNESS : 0, NetMsgType::TX, (*mi).second);
                         pushed = true;
                     }
                 }
-                if (!pushed && inv.type == MSG_TX) {
+                if (!pushed && inv.type == MSG_TX || inv.type == MSG_WITNESS_TX) {
```

Comment:
heh.  doesn't the && bind tighter than the ||?
I think you can just get rid of checking the inv.type, its checked in the containing `else if`